libgenic/Makefile

```diff
@@ -26,7 +26,7 @@ all : libgenic.a
 .objs/test_%: tests/test_%.c .objs/%.o ../tests/stub.c ../tests/cmocka.c ../libgadget/libgadget-utils.a ../libgadget/libgadget.a
 	$(MPICC) $(CFLAGS) -I../tests/ $^ $(LIBS) -o $@
 
-build-tests: $(TESTBIN)
+build-tests: $(TESTBIN) libgenic.a
```

Comment:
I don't see any tests linking against libgenic.a. If they do then the dependency shall be added to `.objs/tests_%` If they don't then we are not even testing it?

Reply:
It's not to link against libgenic.a: it's just to ensure that the build process rebuilds all the object files before running the tests. Despite the % rule it wasn't doing this properly for me.

Reply:
Could it be because libgenic.a wasn't declared in the `%` rule?

Reply:
Probably - do you think that is a better fix? If I did that it would link libgenic.a, but that probably doesn't matter

Reply:
Move libgenic.a to the % rule above?

Reply:
hmm, that doesn't actually work (I think I do not understand make):

[spb43]codes/MP-Gadget/libgenic (gcthread)% make clean
rm -rf .objs libgenic.a
[spb43]codes/MP-Gadget/libgenic (gcthread)% make test
make: *** No rule to make target '.objs/test_power', needed by 'build-tests'.  Stop.
[spb43]codes/MP-Gadget/libgenic (gcthread)%

Reply:
But now the old fix doesn't work either...